File: cli/src/Message.js
export class Message {
  static fromJSON (buffer) {
    return new Message(JSON.parse(buffer.toString()))
  }

  constructor ({ username, command, contents, timestamp, error}) {
    this.username = username
    this.command = command
    this.contents = contents
    this.timestamp = timestamp
    this.error = error
  }

  toJSON () {
    return JSON.stringify({
      username: this.username,
      command: this.command,
      contents: this.contents,
      timestamp: this.timestamp,
      error : this.error
    })
  }

  toString () {
    return this.contents
  }
}

File: src/main/java/com/cooksys/assessment/model/Message.java
package com.cooksys.assessment.model;

public class Message {

	private String username;
	private String command;
	private String contents;
	private String timestamp;
	private boolean error;

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getCommand() {
		return command;
	}

	public void setCommand(String command) {
		this.command = command;
	}

	public String getContents() {
		return contents;
	}

	public void setContents(String contents) {
		this.contents = contents;
	}

	public String getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(String timestamp) {
		this.timestamp = timestamp;
	}

	public boolean isError() {
		return error;
	}

	public void setError(boolean error) {
		this.error = error;
	}
	
	
	
	

}
